File: src/main/java/ac/dnd/server/admission/domain/ApplicantValidator.java
package ac.dnd.server.admission.domain;

import java.time.LocalDateTime;

import org.springframework.stereotype.Component;

import ac.dnd.server.admission.domain.model.ApplicantData;
import ac.dnd.server.admission.exception.ApplicantViewablePeriodEndException;

@Component
public class ApplicantValidator {

	public void viewableValidator(
		final ApplicantData applicant,
		final LocalDateTime now
	) {
		if (!applicant.isViewable(now)) {
			throw new ApplicantViewablePeriodEndException();
		}
	}
}

File: src/main/kotlin/ac/dnd/server/admission/domain/ApplicantValidator.kt
package ac.dnd.server.admission.domain

import ac.dnd.server.admission.domain.model.ApplicantData
import ac.dnd.server.admission.exception.ApplicantViewablePeriodEndException
import org.springframework.stereotype.Component
import java.time.LocalDateTime

@Component
class ApplicantValidator {

    fun viewableValidator(applicant: ApplicantData, now: LocalDateTime) {
        if (!applicant.isViewable(now)) {
            throw ApplicantViewablePeriodEndException()
        }
    }
}